src/store/modules/picklist/actions.ts

```diff
@@ -31,6 +49,46 @@ const actions: ActionTree<PicklistState, RootState> = {
     }
   },
   
+  /**
+   * Get completed picklists
+   */
+  async fetchCompletedPickLists({ commit, dispatch, state }, payload) {
+    let resp;
+    const params = {
+      "inputFields": {
+        "statusId": "PICKLIST_COMPLETED",
+        "facilityId": this.state.user.currentFacility.facilityId,
+      },
+      "orderBy": "picklistDate DESC",
+      "viewSize": 10,
+      "fieldList": ["picklistId", "picklistDate", "statusId", "partyId"],
+      "entityName": "PicklistAndRole",
+      "noConditionFind": "Y"
+    } as any
+
+    if (state.filters.showMyPicklists) params.inputFields.partyId = payload.partyId
+
+    try {
+      resp = await PicklistService.getPicklists(params);
+      if (resp.status === 200 && !hasError(resp) && resp.data.docs?.length > 0) {
+        let list = resp.data.docs;
+        const pickersPartyIds = [...new Set(list.map((item: any) => item.partyId))]
+        const pickersDetails = await dispatch('party/getPickersDetails', pickersPartyIds, { root: true });
+
+        list = list.map((item: any) => ({ ...item, pickersFullName: pickersDetails[item.partyId].fullName }))
+
+        commit(types.PICKLISTS_COMPLETED_UPDATED, { list: list, total: resp.data.count });
+        return resp.data;
+      } else {
+        commit(types.PICKLISTS_COMPLETED_UPDATED, { list: [], total: 0 });
```

Comment:
We should initialise default values, and have single commit at last with response values. If it fails, initial value will be set at last